notebook/services/contents/filemanager.py

```diff
@@ -284,7 +286,9 @@ def _dir_model(self, path, content=True):
                         self.log.warning("Error stat-ing %s: %s", os_path, e)
                     continue
 
-                if not stat.S_ISREG(st.st_mode) and not stat.S_ISDIR(st.st_mode):
+                if (not WINDOWS_CONTAINER
```

Comment:
` os.stat(path)` would throw a `FileNotFound` error. You'd need to either use `lstat` or pass the `follow_symlinks=False` arg (py3 only).

`lstat` gives a 41471 mode:
```python
[20]: st = os.stat(r"C:\jupyter\notebooks", follow_symlinks=False)
... : st
Out[20]: os.stat_result(st_mode=41471, st_ino=281474976710694, st_dev=113970060, st_nlink=1, st_uid=0, st_gid=0, st_size=0, st_atime=1497345137, st_mtime=1497345137, st_ctime=1497345137)

[21]: st.st_file_attributes
Out[21]: 1040
```

If we don't care about checking if a symlink is broken we could just globally use `lstat`. The only other issue is then the check for a *regular* file:
```python
if not stat.S_ISREG(st.st_mode) and not stat.S_ISDIR(st.st_mode):
    self.log.debug("%s not a regular file", os_path)
    continue
```
In that case doesn't recognise the mapped volume as a directory but sees it as a link instead:
```python
In [2]: stat.S_ISREG(41471)
Out[2]: False

In [3]: stat.S_ISDIR(41471)
Out[3]: False

In [4]: stat.S_ISLNK(41471)
Out[4]: True
```
...so the `if not WINDOWS_CONTAINER` could be replaced with `and not stat.S_ISLNK(st.st_mode)`

Happy to go ahead and make those changes if you think that's the right way forward...